ehcache-core/src/main/java/org/ehcache/core/PrefixLogger.java

```diff
@@ -0,0 +1,329 @@
+package org.ehcache.core;
+
+import org.slf4j.Logger;
+import org.slf4j.Marker;
+
+public class PrefixLogger implements Logger {
+
+  private final Logger delegate;
+  private final String prefix;
+  private final String formatEscapedPrefix;
+
+  public PrefixLogger(Logger logger, String prefix) {
+    this.delegate = logger;
+    this.prefix = prefix;
+    this.formatEscapedPrefix = formatEscape(prefix);
+  }
+
+  private static String formatEscape(String string) {
+    return string
+      .replace("\\\\{}", "{}")
+      .replace("\\{}", "{}")
```

Comment:
These first two replaces are wrong. they will turn: "foo\\{}" (and similar) which don't expect a variable sub in to "foo{}" which does.

Reply:
corrected this